radicle-node/src/service.rs

```diff
@@ -630,14 +637,15 @@ where
         // Inbound connection attempt.
     }
 
-    pub fn attempted(&mut self, id: NodeId, addr: &Address) {
-        debug!(target: "service", "Attempted connection to {id} ({addr})");
+    pub fn attempted(&mut self, nid: NodeId, addr: &Address) {
+        debug!(target: "service", "Attempted connection to {nid} ({addr})");
 
-        let persistent = self.config.is_persistent(&id);
-        self.sessions
-            .entry(id)
-            .and_modify(|sess| sess.to_connecting())
-            .or_insert_with(|| Session::connecting(id, persistent, self.rng.clone()));
+        if let Some(sess) = self.sessions.get_mut(&nid) {
+            sess.to_attempted();
+        } else {
+            #[cfg(debug_assertions)]
+            panic!("Service::attempted: unknown session {nid}@{addr}");
+        }
```

Comment:
let () else {}?
```
let Some(sess) = self.sessions.get_mut(..) else {
          #[cfg(debug_assertions)]
            panic!("Service::attempted: unknown session {nid}@{addr}");
}
sess.to_attempted();
```

Reply:
`sess` won't be in scope then because it would need to be returned in the `else`

Reply:
Yeah indeed, was thinking we need a macro that panics in debug mode and logs in release mode..